torch/_inductor/graph.py

```diff
@@ -464,6 +464,8 @@ def __init__(
         # Below field is related to printing debug intermediate tensor values info for debugging
         self.all_codegen_kernel_names: OrderedSet[str] = OrderedSet()
 
+        self.comm_buffer_type: Dict[str, str] = {}
```

Comment:
I do wonder whether this should be on Layout or the graph... I think I have a weak preference for this to be a Layout subclass.

cc: @eellison and @jansel for opinions

Reply:
I agree a layout seems better.